### ISSUE TITLE
Android callouts are cut off and do not show content

### ISSUE THREAD

**Issue by kloovin112358** [react-native-maps/react-native-maps#5216]
### Summary

Callout text and media on Android are cut off, and do not display. Styling changes do not affect how it displays.

### Reproducible sample code

```TSX
import React from "react"
import MapView, { Marker,Callout } from 'react-native-maps';
import { Text, View } from 'react-native';

export default function MyMap() {
   return (
      <MapView
        style={{
            width: '100%',
            height: '100%',
          
        }}
        initialRegion={{
          latitude: 41.8781,
          longitude: -87.6298,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }}
      >
        <Marker
            key={1}
            coordinate={{
                "latitude": 41.884782,
                "longitude": -87.624961
            }}
          >
              <Callout>
                  <View><Text>Some text here</Text></View>
              </Callout>
          </Marker>
      </MapView>
    )
}
```

### Steps to reproduce

Run example on Expo SDK 52.0.6 with react-native-maps version 1.18.0 (latest recommended Expo version).

View on Android, and see callout as screenshotted below.

### Expected result

Callout should show text content.

### Actual result

![Screenshot_20241114_163512_Expo Go](https://github.com/user-attachments/assets/893808d9-53d3-483b-80c4-718bba2bafc2)

See screenshot. Top is cut off of callout.

### React Native Maps Version

1.18.0

### What platforms are you seeing the problem on?

Android

### React Native Version

0.76.1

### What version of Expo are you using?

Not using Expo

### Device(s)

Samsung A15 5G

### Additional information

This was not happening before updating to Expo SDK 52.

**Comment by ayushh2k** [comment]
the same issue occurs for a custom marker too, also copy of [#5165](https://github.com/react-native-maps/react-native-maps/issues/5165)

**Comment by JaRoMasterDev** [comment]
Are there any updates on this? I am experiencing the same issue. Could this possibly be related to Expo?

**Comment by ayushh2k** [comment]
@JaRoMasterDev it is related to the new architecture of react native and might take some time to get fixed, downgrading my react native version fixed it for me

**Comment by JaRoMasterDev** [comment]
That seems like a huge issue. I don't really want to downgrade the react native version. I'll probably try to find a workaround or build my own solution

**Comment by i-umayr** [comment]
@JaRoMasterDev let me know if you find something, facing the same problem

**Comment by JaRoMasterDev** [comment]
I'll definitely let you know if I get to achieve anything useful. But this would probably just be a solution for my specific usecase. I'll have to see what's possible. @ayushh2k Thanks so far, without you I'd probably still be looking for solutions that (apparently) don't exist

**Comment by ayushh2k** [comment]
No problem, I've tried solutions mentioned in other issues like changing the google render to legacy and applying a patch from [#5103](https://github.com/react-native-maps/react-native-maps/pull/5103) but they didn't work, so I had to downgrade to react native 0.74.5 which fixed the issue for me.

**Comment by JaRoMasterDev** [comment]
I also tried several solutions (except downgrading) but none of them seemed to work. Becuase there probably won't be a fix to this in the near future and it'll take even more time until this fix is in the recommended version for Expo, I came up with a solution for myself (@i-umayr ).

I have already added a bottom sheet that displays the same data as my map, so I linked the map to the bottom sheet and the FlatList inside that bottom sheet, to achieve the following behavoir:

If a user klicks on a marker:

- The bottomsheet get's opened to take up some space at the bottom of the screen
- The FlatList get's scrolled to the related item
- The item get's highlighted with a pulsing background

I know this is a "fix" that only fits my specific needs, but I guess this works for me. I might even like it a bit more than using the normal Callouts.


https://github.com/user-attachments/assets/8af64c76-ae1e-4104-8b75-6d6cd2180e99

**Comment by i-umayr** [comment]
Thank you so much for the detailed response @JaRoMasterDev

**Comment by ewikum** [comment]
Hey Folks, 

In your `android/gradle.properties`, chnage `newArchEnabled=false` to `newArchEnabled=false`. It will fix this for now. I tested on `0.76.3`. Thank me later ! 😀

**Comment by ihsanktmr** [comment]
the same issue here

**Comment by ChanuPerera** [comment]
> Hey Folks,
> 
> In your `android/gradle.properties`, chnage `newArchEnabled=true` to `newArchEnabled=false`. It will fix this for now. I tested on `0.76.3`. Thank me later ! 😀

This is the correct solution , if you are looking a way to go through this issue without downgrading the react native version. This worked for me.

**Comment by TheGuyWithoutH** [comment]
Is there a way to fix it without changing the new architecture flag?

**Comment by david-fiaty** [comment]
+1

**Comment by trdp30** [comment]
This is one of Maps' critical functionalities. It would be helpful if it could be prioritized.

**Comment by JamesOrtizBrazil** [comment]
This is one of Maps' critical functionalities. It would be helpful if it could be prioritized.

**Comment by lauz1n** [comment]
This is one of Maps' critical functionalities. It would be helpful if it could be prioritized.

**Comment by LukasAlencar** [comment]
I have the same problem

**Comment by luzhnyak** [comment]
I have the same problem
"expo": "^52.0.28"
"react-native": "0.76.6",
"react-native-maps": "1.18.0",

**Comment by Shmublon** [comment]
Same here
"react-native-maps": "^1.20.1",
"react": "18.3.1",
"react-native": "0.77.0",

**Comment by anasbehhari-harmony** [comment]
Same issue for me

**Comment by yavoo** [comment]
+1

**Comment by avinash797** [comment]
+1

**Comment by AntonioHReyes** [comment]
+1

**Comment by AbdurakhmonZiyodov** [comment]
+1

**Comment by manuelduarte077** [comment]
Sigue pasando este error

**Comment by abhion** [comment]
any update on this?

**Comment by AFDHAL2009** [comment]
import MapView, { Callout, Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import React, { useState } from 'react';

import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TouchableOpacity,
} from 'react-native';

export default function App() {
  const [mapLat, setMapLat] = useState(6.841776681);
  const [mapLong, setMapLong] = useState(79.869319);
  const locationData = [
    { latitude: 6.841776681, longitude: 79.869319 ,weight:1,},
    { latitude: 6.84076664, longitude: 79.871323 ,weight:2,},
  ];
  return (
    <View style={styles.container}>
      <MapView
        style={styles.map}
        initialRegion={{
          latitude: mapLat,
          longitude: mapLong,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }}>
        {locationData.map((data, index) => (
          <Marker
            key={index}
            coordinate={{
              latitude: data.latitude,
              longitude: data.longitude,
            }}
            title={`Marker ${index + 1}`}
            description={`Weight: ${data.weight}`}
             onPress={() => {
            alert(data.weight);
          }}
          tracksViewChanges={true} 
            >
            <TouchableOpacity
              style={{
              
              
               //borderRadius: 15,
                elevation: 2,
               borderWidth: 1.0,
                borderColor: 'gray',
        
              
              }}>
              <Text
               
               >
             { "custom text"}
              </Text>
            </TouchableOpacity>
          </Marker>
        ))}
      </MapView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  map: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  },
});

![Image](https://github.com/user-attachments/assets/c734746e-7aff-42ce-8b81-d07f591db846)

**Comment by salah-ghanim** [comment]
This has been fixed in the alpha release, will be merged into master later, but it will require Fabric to be enabled / new arch

**Comment by JamesOrtizBrazil** [comment]
> This has been fixed in the alpha release, will be merged into master later, but it will require Fabric to be enabled / new arch

great news!

**Comment by muhammadtaha1998** [comment]
> This has been fixed in the alpha release, will be merged into master later, but it will require Fabric to be enabled / new arch

Could you please show me the fix so I can patch it myself till it is released?

**Comment by abhion** [comment]
@salah-ghanim I'm using the alpha version and I get this error

`Error: Unsupported top level event type "topMapLoaded" dispatched`

**Comment by abhimvp** [comment]
+1 same issue , any updates ?

**Comment by manuelduarte077** [comment]
Todo sigue igual, podes usar algun poper o crear tu propio alert

**Comment by yash-appwizards** [comment]
Callout is Being Cut Off or Not Displaying Properly on Android

Description
I'm using react-native-maps to display custom markers with a Callout. The Callout works perfectly on iOS, but on Android, it gets cut off or does not display properly. Sometimes, it is partially visible, and other times it doesn't appear at all.

<MapView
                key={Math.random()}
                ref={mapRef}
                style={styles.map}
                provider={PROVIDER_GOOGLE}
                initialRegion={initialRegion}
                showsUserLocation={true}
                showsMyLocationButton={true}
                showsCompass={true}
                scrollEnabled={true}
                zoomEnabled={true}
                pitchEnabled={true}
                rotateEnabled={true}
                moveOnMarkerPress={false}
                toolbarEnabled={true}
            >
                {careTakers.map((marker) => (
                    marker.address && (
                        <Marker
                            key={marker.id}
                            coordinate={{
                                latitude: marker.address.latitude,
                                longitude: marker.address.longitude
                            }}
                            tracksViewChanges={true}
                            onPress={() => {
                                mapRef.current?.animateToRegion({
                                    latitude: marker.address.latitude,
                                    longitude: marker.address.longitude,
                                    latitudeDelta: 0.01,
                                    longitudeDelta: 0.01,
                                }, 1000);
                            }}
                        >
                            <View style={styles.markerContainer} renderToHardwareTextureAndroid={true}>
                                <Icon name="youtube" size={RFValue(12)} color={Colors.PRIMARY} />
                            </View>
                            <Callout
                                renderToHardwareTextureAndroid={true}
                                onPress={onPress}
                            >
                                <TouchableOpacity activeOpacity={0.7}>
                                    {renderMarkerCallout(marker)}
                                </TouchableOpacity>
                            </Callout>
                        </Marker>
                    )
                ))}
            </MapView>

**Comment by jamesattard** [comment]
Can confirm having the same issue with the callout being cut off. Any idea when this will be merged to master as it seems to have been going for quite a long time now

**Comment by miasK3011** [comment]
Still no fix?

**Comment by TehDrink** [comment]
> Can confirm having the same issue with the callout being cut off. Any idea when this will be merged to master / expo?

Ditto, having npm installed the latest alpha [1.21.0-alpha.112] makes the markers disappear :(

**Comment by jamesattard** [comment]
Just in case some people are still struggling, I ended up using modals to open when the marker is clicked. It is a clean solution and personally, I prefer it even better than the callouts.

**Comment by Gabrielmmk** [comment]
Still no fix?

**Comment by Gabrielmmk** [comment]
> [@Gabrielmmk](https://github.com/Gabrielmmk) Are you still using a react-native-maps version < 1.21.0? If you are using the new arch, you should use at least 1.21.0, but I'd recommend 1.23.8. It's the newest version and fixes many bugs

I'll test it and come back here to say if it worked for me.

**Comment by reyesdiego** [comment]
The version 1.23.8 throws another error
 ERROR  Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'RNMapsAirModule' could not be found. Verify that a module by this name is registered in the native binary., js engine: hermes

because of the versions Im using it means it does not allow me to use anything more than 1.20

**Comment by demiters** [comment]
> The version 1.23.8 throws another error ERROR Invariant Violation: TurboModuleRegistry.getEnforcing(...): 'RNMapsAirModule' could not be found. Verify that a module by this name is registered in the native binary., js engine: hermes
> 
> because of the versions Im using it means it does not allow me to use anything more than 1.20

I'm getting this with react-native-maps@1.24.5, got rid of the error for now by removing usage of CalloutSubview.

**Comment by TableTennisCoder** [comment]
Use "CalloutSubview" inside the "Callout". Then it works on android as well


import MapView, {
  Callout,
  CalloutSubview,....
} from "react-native-maps";


 `<Callout onPress={() => router.push(`/cinema/${cinema.iD}`)}>
      <CalloutSubview>
<<Your content here >>
       </CalloutSubview>
  </Callout>`

**Comment by wellataide** [comment]
i've got solving this by updating my packages, mainly react-native-maps to `1.24.3`

```
...
"expo": "53.0.19",
"react-native": "0.79.5",
"react-native-maps": "1.24.3",
...
```

**Comment by icarosun** [comment]
Same error: 

```
...
"expo": "53.0.20",
"react-native": "0.79.5",
"react-native-maps": "1.20.1"
...
```

**Comment by AlejandroRM-DEV** [comment]
> "expo": "53.0.19",
> "react-native": "0.79.5",
> "react-native-maps": "1.24.3",

It didn't work for me, do you remember if you did something else?

![Image](https://github.com/user-attachments/assets/aac1ddb3-3bd6-4876-887e-69c8a5ba31c2)

**Comment by stavbi1** [comment]
any updates on this bug?

**Comment by Rk5422686** [comment]
If you prefer not to disable the new architecture or upgrade react-native-maps to the latest version, here’s a solution that worked for me.

My current setup in package.json:
```
    "react": "^18.3.1",
    "react-native": "^0.77.1",
    "react-native-maps": "^1.20.1",
```
I applied the following patch to fix the issue with custom markers being cut off:

```
diff --git a/android/src/main/java/com/rnmaps/maps/MapMarker.java b/android/src/main/java/com/rnmaps/maps/MapMarker.java
index b755f9b..a9f6442 100644
--- a/android/src/main/java/com/rnmaps/maps/MapMarker.java
+++ b/android/src/main/java/com/rnmaps/maps/MapMarker.java
@@ -448,6 +448,14 @@ public class MapMarker extends MapFeature {
     updateTracksViewChanges();
   }
 
+
+  @Override
+  protected void onLayout(boolean changed, int l, int t, int r, int b) {
+    super.onLayout(changed, l, t, r, b);
+    this.height = b-t;
+    this.width = r-l;
+  }
+
   private BitmapDescriptor getIcon() {
     if (hasCustomMarkerView) {
       // creating a bitmap from an arbitrary view
```
I've created a patch that you can apply directly— enjoy :)

**Comment by salah-ghanim** [comment]
closing this issue, custom callouts are working as expected in android examples.

if you still have an issue with this please submit a new bug report with a reproducible repository and expected vs observed reporting.

**Comment by bhavesh-trentium** [comment]
@salah-ghanim @ewikum Hey guys, I noticed that most of the `react native maps marker not showing on Android` issues are just being closed with the patch solution `newArchEnabled=false`.

But that’s not really a fix — it’s just a workaround. In my case, disabling the new architecture actually breaks other packages in the project.

Would be great if we could have a proper fix instead of just closing issues with that temporary solution.

**Comment by demiters** [comment]
Have to say I'm also still semi blocked by this due to the workaround affecting other parts of my project.

**Comment by aelaboussi** [comment]
Resolved by upgrading to `1.23.12`, i was using `1.20.1` (RN 0.79.5 and Expo 53).

**Comment by Ganesh-Rana-tech** [comment]
> ### Summary
> Callout text and media on Android are cut off, and do not display. Styling changes do not affect how it displays.
> 
> ### Reproducible sample code
> import React from "react"
> import MapView, { Marker,Callout } from 'react-native-maps';
> import { Text, View } from 'react-native';
> 
> export default function MyMap() {
>    return (
>       <MapView
>         style={{
>             width: '100%',
>             height: '100%',
>           
>         }}
>         initialRegion={{
>           latitude: 41.8781,
>           longitude: -87.6298,
>           latitudeDelta: 0.0922,
>           longitudeDelta: 0.0421,
>         }}
>       >
>         <Marker
>             key={1}
>             coordinate={{
>                 "latitude": 41.884782,
>                 "longitude": -87.624961
>             }}
>           >
>               <Callout>
>                   <View><Text>Some text here</Text></View>
>               </Callout>
>           </Marker>
>       </MapView>
>     )
> }
> ### Steps to reproduce
> Run example on Expo SDK 52.0.6 with react-native-maps version 1.18.0 (latest recommended Expo version).
> 
> View on Android, and see callout as screenshotted below.
> 
> ### Expected result
> Callout should show text content.
> 
> ### Actual result
> ![Screenshot_20241114_163512_Expo Go](https://github.com/user-attachments/assets/893808d9-53d3-483b-80c4-718bba2bafc2)
> 
> See screenshot. Top is cut off of callout.
> 
> ### React Native Maps Version
> 1.18.0
> 
> ### What platforms are you seeing the problem on?
> Android
> 
> ### React Native Version
> 0.76.1
> 
> ### What version of Expo are you using?
> Not using Expo
> 
> ### Device(s)
> Samsung A15 5G
> 
> ### Additional information
> This was not happening before updating to Expo SDK 52.

getting same issue 
on ios callout is perfectly showing ui, but on android it is not showing

**Comment by Ganesh-Rana-tech** [comment]
@kloovin112358 have you figured out any solution

**Comment by Perseus24** [comment]
Using a bottom sheet or a modal may be a good workaround depending on what the callout should contain. Fortunately for me, my design already included a bottom sheet which is kinda better than showing a callout on my specific project.

**Comment by Ganesh-Rana-tech** [comment]
@Perseus24 thanks for your insights , i solved this issue by creating my custom overlay anchored above the selected marker.
